app/models/host_aspect_base.rb

```diff
@@ -0,0 +1,36 @@
+class HostAspectBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  def self.inherited(subclass)
+    super
+    subclass.class_eval do
+      belongs_to_host
+    end
+  end
+
+  def info
```

Comment:
I'll add comments with explanation what each method is used for, sure.
They should not throw exceptions, because you don't have to override them if you don't have what to add to the base method. The method will be called for each aspect, so if one of them throws an exception - the whole method will fail.